lms/static/scripts/frontend_apps/components/test/BookPicker-test.js

```diff
@@ -0,0 +1,142 @@
+import { mount } from 'enzyme';
+import { createElement } from 'preact';
+import { act } from 'preact/test-utils';
+
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+import { waitForElement } from '../../../test-util/wait';
+import * as bookData from '../../utils/vitalsource-sample-data';
+import BookPicker, { $imports } from '../BookPicker';
+
+describe('BookPicker', () => {
+  let fakeAPI;
+
+  const renderBookPicker = (props = {}) =>
+    mount(<BookPicker authToken="dummy-auth-token" {...props} />);
+
+  beforeEach(() => {
+    fakeAPI = {
+      fetchBooks: sinon.stub().resolves(bookData.bookList),
+      fetchChapters: sinon
+        .stub()
+        .callsFake(async (authToken, bookID) => bookData.chapterData[bookID]),
+    };
+
+    $imports.$mock(mockImportedComponents());
+    $imports.$restore({
+      './Dialog': true,
+    });
+    $imports.$mock({
+      '../utils/api': fakeAPI,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  const selectBook = wrapper => {
+    const bookList = wrapper.find('BookList');
+    const book = bookList.prop('books')[0];
+    act(() => {
+      bookList.prop('onSelectBook')(book);
+    });
+    wrapper.update();
+    return book;
+  };
+
+  const selectChapter = wrapper => {
+    const chapterList = wrapper.find('ChapterList');
+    const chapter = chapterList.prop('chapters')[0];
+    act(() => {
+      chapterList.prop('onSelectChapter')(chapter);
+    });
+    wrapper.update();
+    return chapter;
+  };
+
+  const clickSelectButton = wrapper => {
+    act(() => {
+      wrapper
+        .find('LabeledButton[data-testid="select-button"]')
+        .prop('onClick')();
+    });
+    wrapper.update();
+  };
+
+  it('fetches and displays book list when picker is opened', async () => {
+    const picker = renderBookPicker();
+
+    let bookList = picker.find('BookList');
+    assert.deepEqual(bookList.prop('books'), []);
```

Comment:
This is specifically testing that the book list is empty when the picker is opened. Is it worthy of its own test, or commenting, or anything?

Reply:
I've added some comments here to highlight the different phases of this test.